components/tikv_util/src/logger/file_log.rs

```diff
@@ -230,12 +193,118 @@ impl Rotator for RotateBySize {
     }
 }
 
+#[derive(Debug)]
+pub enum Task {
+    Archive,
+}
+
+impl Display for Task {
+    fn fmt(&self, f: &mut Formatter<'_>) -> fmt::Result {
+        write!(f, "{:?}", self)
+    }
+}
+
+struct LogInfo {
+    f: DirEntry,
+    dt: DateTime<Local>,
+}
+
+fn dt_from_file_name(path: &Path, prefix: &str) -> Option<DateTime<Local>> {
+    let file_name_with_dt = path.file_stem().unwrap();
+    let mut dt = file_name_with_dt
+        .to_str()
+        .unwrap()
+        .to_string()
+        .replace(prefix, "");
+    if let Some(offset) = dt.find('T') {
+        unsafe {
+            // -2021-12-13T16-08-27.621 => -2021-12-13T16:08:27.621
+            let dt = dt.as_bytes_mut();
+            if dt.len() < offset + 6 {
+                return None;
+            }
+            dt[offset + 3] = b':';
+            dt[offset + 6] = b':';
+        };
```

Comment:
The code extracting time looks unsound. It can make the program panic in L230 if you modify the some bytes to `:` causing the string invalid UTF-8.

A safe way is changing it to raw bytes, and then convert it to string again...

Reply:
Actually, I think we can use the `parse_from_str` method provided by `chrono` directly.

Reply:
You are right, it has been modified, much better than my own parser, thanks!